src/tests/Common/Directory.Build.targets

```diff
@@ -34,7 +35,47 @@
   </ItemDefinitionGroup>
 
   <Target Name="CopyDependencyToCoreRoot"
-          DependsOnTargets="ResolveAssemblyReferences;ResolveRuntimeFilesFromLocalBuild">
+    DependsOnTargets="ResolveAssemblyReferences;ResolveRuntimeFilesFromLocalBuild">
+
+    <!-- Copy apphost to crossgen2_publish directory.
+        The default configuration between product and tests are flipped. When we build project like:
+        `build -c debug -rc checked`, the corresponding test command is `src/tests/build -checked -p:LibrariesConfiguration=debug`,
+        instead of `-debug -p:RuntimeConfiguration=checked`. That forces us to either pass the
+        `HostConfiguration=debug` explicitly or fix this disparity; both of which will break the dev workflow.
+        
+        As a workaround, we will first check if the directory pointed by `HostConfiguration` exists, then check
+        `LibrariesConfiguration`.
+    -->
```

Comment:
How will this workaround handle a local workflow like `build clr+libs -c debug -lc release`? Will it still work correctly? Or do we need to also check the apphost built with `RuntimeConfiguration`?

Reply:
Yup, it’s basically tested in the CI as well. What happens is that HostConfiguration is resolved to debug (because the un-prefixed Configuration is debug). But we then build the tests with opposite `src/tests/build.sh release -LibrariesConfiguration=Debug`, which requires this reassignments (because now HostConfiguration will be resolved to release instead of debug). As you mentioned above, once we integrate test build script with top-level build.sh/cmd, we can remove this workaround then.